src/libawkward/array/NumpyArray.cpp

```diff
@@ -651,16 +651,58 @@ namespace awkward {
     }
   }
 
+  const std::vector<ssize_t> flatten_shape(const std::vector<ssize_t>& shape, int64_t axis) {
+    if (shape.size() == 1) {
+      return std::vector<ssize_t>();
+    }
+    else {
+      std::vector<ssize_t> out;
+      const auto& indx = begin(shape) + axis;
+      if (indx > begin(shape)) {
+        out.insert(end(out), begin(shape), indx);
+      }
+      out.emplace_back(shape[axis]*shape[axis + 1]);
+      out.insert(end(out), indx + 2, end(shape));
+      return out;
+    }
+  }
+
+  const std::vector<ssize_t> flatten_strides(const std::vector<ssize_t>& strides, int64_t axis) {
+    if (strides.size() == 1) {
+      return std::vector<ssize_t>();
+    }
+    else {
+      std::vector<ssize_t> out;
+      const auto indx = begin(strides) + axis;
```

Comment:
and here

Reply:
fixed in https://github.com/scikit-hep/awkward-1.0/pull/98